tests/sentry/api/endpoints/test_organization_projects.py

```diff
@@ -1,167 +1,170 @@
 
-from exam import fixture
+from django.urls import reverse
 
 from sentry.models import ApiKey
 from sentry.testutils import APITestCase
 
 
-class OrganizationProjectsTest(APITestCase):
-    @fixture
-    def path(self):
-        return f"/api/0/organizations/{self.organization.slug}/projects/"
+class OrganizationProjectsTestBase(APITestCase):
+    endpoint = "sentry-api-0-organization-projects"
 
-    def check_valid_response(self, response, expected_projects):
-        assert response.status_code == 200, response.content
+    @staticmethod
+    def check_valid_response(response, expected_projects):
         assert [project.id for project in expected_projects] == [
             int(project_resp["id"]) for project_resp in response.data
         ]
 
-    def test_simple(self):
+    def test_api_key(self):
+        key = ApiKey.objects.create(organization=self.organization, scope_list=["org:read"])
+
+        project = self.create_project(teams=[self.team])
+
+        path = reverse(self.endpoint, args=[self.organization.slug])
+        response = self.client.get(
+            path,
+            HTTP_AUTHORIZATION=b"Basic " + b64encode(f"{key.key}:".encode("utf-8")),
+        )
+        self.check_valid_response(response, [project])
+
+
+class OrganizationProjectsTest(OrganizationProjectsTestBase):
+    def setUp(self):
+        super().setUp()
         self.login_as(user=self.user)
+
+    def test_simple(self):
         project = self.create_project(teams=[self.team])
 
-        response = self.client.get(self.path)
+        response = self.get_valid_response(self.organization.slug)
         self.check_valid_response(response, [project])
         assert self.client.session["activeorg"] == self.organization.slug
 
     def test_with_stats(self):
-        self.login_as(user=self.user)
-
         projects = [self.create_project(teams=[self.team])]
 
-        response = self.client.get(f"{self.path}?statsPeriod=24h", format="json")
+        response = self.get_valid_response(self.organization.slug, qs_params={"statsPeriod": "24h"})
         self.check_valid_response(response, projects)
         assert "stats" in response.data[0]
 
-        response = self.client.get(f"{self.path}?statsPeriod=14d", format="json")
+        response = self.get_valid_response(self.organization.slug, qs_params={"statsPeriod": "14d"})
         self.check_valid_response(response, projects)
         assert "stats" in response.data[0]
 
-        response = self.client.get(f"{self.path}?statsPeriod=", format="json")
+        response = self.get_valid_response(self.organization.slug, qs_params={"statsPeriod": ""})
         self.check_valid_response(response, projects)
         assert "stats" not in response.data[0]
 
-        response = self.client.get(f"{self.path}?statsPeriod=48h", format="json")
-        assert response.status_code == 400
+        self.get_valid_response(
+            self.organization.slug, qs_params={"statsPeriod": "48h"}, status_code=400
```

Comment:
Could `qs_params` be `data` or `query` to align with the `self.client` API. Having to remember different names can be taxing.

Reply:
The problem is `**params` are directly transformed to the POST data so we can't use any good names without collisions with the keys we want to test. 
Also, I agree that different names is a mentally taxing hurdle, I keep getting `status_code` and `response_code` mixed up when I write tests.